x-pack/plugins/security_solution/cypress/e2e/timelines/creation.cy.ts

```diff
@@ -142,17 +145,66 @@ describe('Timelines', (): void => {
         .should('have.text', getTimeline().notes);
     });
 
-    it('should update timeline after adding eql', () => {
-      cy.intercept('PATCH', '/api/timeline').as('updateTimeline');
-      const eql = 'any where process.name == "zsh"';
-      addEqlToTimeline(eql);
+    describe('correlation tab', () => {
+      it('should update timeline after adding eql', () => {
+        cy.intercept('PATCH', '/api/timeline').as('updateTimeline');
+        const eql = 'any where process.name == "zsh"';
+        addEqlToTimeline(eql);
 
-      cy.wait('@updateTimeline', { timeout: 10000 }).its('response.statusCode').should('eq', 200);
+        cy.wait('@updateTimeline', { timeout: 10000 }).its('response.statusCode').should('eq', 200);
 
-      cy.get(`${TIMELINE_TAB_CONTENT_EQL} ${SERVER_SIDE_EVENT_COUNT}`)
-        .invoke('text')
-        .then(parseInt)
-        .should('be.gt', 0);
+        cy.get(`${TIMELINE_TAB_CONTENT_EQL} ${SERVER_SIDE_EVENT_COUNT}`)
+          .invoke('text')
+          .then(parseInt)
+          .should('be.gt', 0);
+      });
+
+      describe('updates', () => {
+        const eql = 'any where process.name == "zsh"';
+        beforeEach(() => {
+          cy.intercept('PATCH', '/api/timeline').as('updateTimeline');
+          addEqlToTimeline(eql);
+          // TODO: It may need a further refactor to handle the frequency with which react calls this api
+          // Since it's based on real time text changes...and real time query validation
+          // there's almost no guarantee on the number of calls, so a cypress.wait may actually be more appropriate
+          cy.wait('@updateTimeline');
+          cy.wait('@updateTimeline');
+          cy.reload();
+          cy.get(TIMELINE_CORRELATION_INPUT).should('be.visible');
+          cy.get(TIMELINE_CORRELATION_INPUT).should('have.text', eql);
+        });
+
+        it('should update timeline after removing eql', () => {
+          cy.intercept('PATCH', '/api/timeline').as('updateTimeline');
+          cy.get(TIMELINE_CORRELATION_INPUT).should('be.visible');
+          waitForTimelineChanges();
+          cy.get(TIMELINE_CORRELATION_INPUT).type('{selectAll} {del}').clear();
+          // TODO: It may need a further refactor to handle the frequency with which react calls this api
+          // Since it's based on real time text changes...and real time query validation
+          // there's almost no guarantee on the number of calls, so a cypress.wait may actually be more appropriate
+          cy.wait('@updateTimeline');
```

Comment:
@PhilippeOberti 😵

Reply:
Alternative would be a cy.wait(time), but I'd like to avoid that where possible.